lib/Driver/ToolChains.cpp

```diff
@@ -1471,6 +1519,222 @@ toolchains::Darwin::constructInvocation(const LinkJobAction &job,
+  getRuntimeStaticLibraryPath(StaticRuntimeLibPath, context.Args, *this);
+
+  // Add the runtime library link path, which is platform-specific and found
+  // relative to the compiler.
+  if (!(staticExecutable || staticStdlib)) {
+    Arguments.push_back("-L");
+    Arguments.push_back(context.Args.MakeArgString(SharedRuntimeLibPath + "/"
+                          + getTriple().getArchName()));
+  }
+
+  // If we're cross-compiling, we likely already have some library paths set
+  // up through environment variables.
+  // Pass them through to the linker here for convenience.
+  const char *VCToolsInstallDir = ::getenv("VCToolsInstallDir");
+  const char *UniversalCRTSdkDir = ::getenv("UniversalCRTSdkDir");
+  const char *UCRTVersion = ::getenv("UCRTVersion");
+
+  StringRef tripleWinArchName;
+  switch (getTriple().getArch()) {
+    case llvm::Triple::x86:
+      tripleWinArchName = "x86";
+      break;
+    case llvm::Triple::x86_64:
+      tripleWinArchName = "x64";
+      break;
+    case llvm::Triple::arm:
+      tripleWinArchName = "arm";
+      break;
+    case llvm::Triple::aarch64:
+      tripleWinArchName = "arm64";
+      break;
+    default:
+      tripleWinArchName = getTriple().getArchName();
+      break;
+  }
+
+  StringRef VCToolsDir(VCToolsInstallDir);
+
+  if (!VCToolsDir.empty()) {
+    Arguments.push_back("-L");
+    Arguments.push_back(context.Args.MakeArgString(llvm::Twine(VCToolsInstallDir) 
+        + "/Lib/" + tripleWinArchName));
+  }
+
+  StringRef UCRTDir(UniversalCRTSdkDir);
+  StringRef UCRTVer(UCRTVersion);
+
+  if (!UCRTDir.empty() && !UCRTVer.empty()) {
+      Arguments.push_back("-L");
+      Arguments.push_back(context.Args.MakeArgString(llvm::Twine(UCRTDir) 
+          + "/Lib/" + llvm::Twine(UCRTVer) + "/ucrt/" + tripleWinArchName));
+      Arguments.push_back("-L");
+      Arguments.push_back(context.Args.MakeArgString(llvm::Twine(UCRTDir) 
+          + "/Lib/" + llvm::Twine(UCRTVer) + "/um/" + tripleWinArchName));
+  }
+
+  addPrimaryInputsOfType(Arguments, context.Inputs, types::TY_Object);
+  addInputsOfType(Arguments, context.InputActions, types::TY_Object);
+
+  for (const Arg *arg : context.Args.filtered(options::OPT_F,
+                                              options::OPT_Fsystem)) {
+    if (arg->getOption().matches(options::OPT_Fsystem))
+      Arguments.push_back("-iframework");
+    else
+      Arguments.push_back(context.Args.MakeArgString(arg->getSpelling()));
+    Arguments.push_back(arg->getValue());
+  }
+
+  if (!context.OI.SDKPath.empty()) {
+    Arguments.push_back("--sysroot");
+    Arguments.push_back(context.Args.MakeArgString(context.OI.SDKPath));
+  }
+
+  // Link the standard library.
+  Arguments.push_back("-L");
+
+  if (staticExecutable) {
+    Arguments.push_back(context.Args.MakeArgString(StaticRuntimeLibPath));
+
+    SmallString<128> linkFilePath = StaticRuntimeLibPath;
+    llvm::sys::path::append(linkFilePath, "static-executable-args.lnk");
```

Comment:
Does this work on Windows?

Reply:
The static stdlib isn't currently building on Windows, but thinking about it more I think we'd end up passing the same arguments here whether we're building static or shared, so this section should be removed. @compnerd, thoughts?

Reply:
I meant linker scripts in general, but that's also a reasonable question.

Reply:
Yeah, I think I made a similar point when I looked through this, that the invocation should be similar.

@jrose-apple linker scripts are not supported on Windows, but this isn't a linker script, it is a response file, which is supported (link and lld both support it, and those are the only viable linkers atm).

Reply:
Whoops, of course. Thanks, Saleem.